Cabal/src/Distribution/PackageDescription/Check.hs

```diff
@@ -846,6 +847,12 @@ data PackageCheck =
      | PackageDistInexcusable { explanation :: CheckExplanation }
   deriving (Eq, Ord)
 
+-- | Would Hackage refuse a package because of this error?
+isHackageDistError :: PackageCheck -> Bool
+isHackageDistError (PackageDistSuspicious {}) = False
+isHackageDistError (PackageDistSuspiciousWarn {}) = False
+isHackageDistError _ = True
```

Comment:
(Without having looked at the `PackageCheck` constructors:)
Can we get the catch-all expanded into the constructors?  (Forces future extensions of `PackageCheck` to revisit this predicate.)

Reply:
Sensible reasoning, added!

Reply:
@andreasabel was that your only suggestion? If so, if you could approve it, I have another PR ready to close #8880.